src/java/main/org/apache/zookeeper/server/quorum/Leader.java

```diff
@@ -1165,6 +1165,58 @@ synchronized public long startForwarding(LearnerHandler handler,
     }
     // VisibleForTesting
     protected final Set<Long> connectingFollowers = new HashSet<Long>();
+
+    private volatile boolean quitWaitForEpoch = false;
+    private volatile long timeStartWaitForEpoch = -1;
+    private volatile SyncedLearnerTracker voteSet;
+
+    public static final String MIN_TIME_WAIT_FOR_EPOCH = "zookeeper.leader.minTimeToWaitForEpoch";
```

Comment:
should comment this. isn't it really MAX_TIME_TO_WAIT_FOR_EPOCH?

Reply:
From the quit decision maker, it means the minimum time we should give to the leader to wait for those epoch proposals.

From the leader perspective, you can say this is the maximum time it can wait for those epoch proposals.

Since this variable is used in the quit decision maker, that's why I called it minTimeToWaitForEpoch. 

Maybe I'm wrong since I'm not native English speaker, I can change it to MAX_TIME_TO_WAIT_FOR_EPOCH if I get another one vote for this name :)

Reply:
MAX_TIME_TO_WAIT_FOR_EPOCH feels more intuitive for me. Also there should be some documentation of this new property in admin doc so users know its existence and how to configure it.

Reply:
Sounds good, I'll rename it to MAX_TIME_TO_WAIT_FOR_EPOCH, and add the documents.